Sources/SwiftSyntax/Trivia.swift

```diff
@@ -69,6 +69,7 @@ public struct Trivia: Sendable {
 
   /// Creates a new ``Trivia`` by merging in the given trivia. Only includes one
   /// copy of a common prefix of `self` and `trivia`.
+  @available(*, deprecated, message: "Use mergingCommonPrefix(trivia) or mergingCommonSuffix(trivia) instead")
   public func merging(_ trivia: Trivia?) -> Trivia {
```

Comment:
What do you think of continuing to have a single `merging` function that merges both a common prefix and a common suffix? I would expect that that’s what most callers would prefer. I doubt that anyone is actively thinking about whether they want to merge the prefix or the suffix?

Reply:
The issue here is `merging` doesn't actually merge by common prefix or common suffix despite its documentation, changing its behavior may break client code that depends on it, that's why I suggested to deprecate it.

Reply:
I would consider that a bug fix and thus an acceptable behavior change.